### PR TITLE
Impl nodeinfo

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: src/iamclient/publicservicehandler.cpp [hunk 1]
```diff
@@ -92,6 +93,49 @@ Error PublicServiceHandler::GetCert(const String& certType, const Array<uint8_t>
...
+{
+    LOG_DBG() << "Getting node info";
+
+    auto ctx = std::make_unique<grpc::ClientContext>();
+    ctx->set_deadline(std::chrono::system_clock::now() + cIAMPublicServiceTimeout);
+
+    iamanager::v5::NodeInfo nodeInfoResponse;
+
+    if (auto status = mStub->GetNodeInfo(ctx.get(), google::protobuf::Empty {}, &nodeInfoResponse); !status.ok()) {
+        return Error(ErrorEnum::eRuntime, status.error_message().c_str());
```
Comment (al1img) [inline src/iamclient/publicservicehandler.cpp:106]: Wrap error

File: src/iamclient/publicservicehandler.cpp [hunk 2]
```diff
@@ -92,6 +93,49 @@ Error PublicServiceHandler::GetCert(const String& certType, const Array<uint8_t>
...
+    }
+
+    if (auto err = pbconvert::ConvertToAos(nodeInfoResponse, nodeInfo); !err.IsNone()) {
+        return err;
```
Comment (al1img) [inline src/iamclient/publicservicehandler.cpp:110]: ditto

File: src/iamclient/permservicehandler.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ * Private
+ **********************************************************************************************************************/
+
+std::shared_ptr<grpc::ChannelCredentials> PermissionsServiceHandler::CreateCredentials()
+{
+    iam::certhandler::CertInfo certInfo;
+
+    auto [credential, err] = mCertProvider->GetMTLSCredentials(mCertStorage.c_str());
+    if (!err.IsNone()) {
+        throw std::runtime_error("Failed to get MTLS config");
```
Comment (al1img) [inline src/iamclient/permservicehandler.cpp:107]: small letter

File: include/iamclient/permservicehandler.hpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    Error GetPermissions(const String& secret, const String& funcServerID, InstanceIdent& instanceIdent,
+        Array<iam::permhandler::PermKeyValue>& servicePermissions) override;
+
+private:
+    static constexpr auto cIAMPermissionsServiceTimeout = std::chrono::seconds(10);
+
+    std::shared_ptr<grpc::ChannelCredentials> CreateCredentials();
+
+    TLSCredentialsItf* mCertProvider {};
```
Comment (al1img) [inline include/iamclient/permservicehandler.hpp:70]: Consider renaming to something like `mTLSCredentials` for example.

File: tests/iamclient/mocks/certprovider.hpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CERTPROVIDER_HPP_
```
Comment (al1img) [inline tests/iamclient/mocks/certprovider.hpp:7]: rename file to "iamclientmock.hpp"

File: tests/iamclient/mocks/certprovider.hpp [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+#define CERTPROVIDER_HPP_
+
+#include <gmock/gmock.h>
+
+#include "iamclient/publicservicehandler.hpp"
+
+namespace testing {
```
Comment (al1img) [inline tests/iamclient/mocks/certprovider.hpp:14]: put into `aos::common::iamclient` namespace

File: tests/iamclient/mocks/certprovider.hpp [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+class MockCertProvider : public aos::common::iamclient::TLSCredentialsItf {
```
Comment (al1img) [inline tests/iamclient/mocks/certprovider.hpp:16]: name `TLSCredentialsMock`

File: src/iamclient/publicservicehandler.cpp (hunk 1 above)
Comment (MykolaSolyanko) [reply src/iamclient/publicservicehandler.cpp]: Done

File: src/iamclient/publicservicehandler.cpp (hunk 2 above)
Comment (MykolaSolyanko) [reply src/iamclient/publicservicehandler.cpp]: Done

File: src/iamclient/permservicehandler.cpp (hunk 3 above)
Comment (MykolaSolyanko) [reply src/iamclient/permservicehandler.cpp]: Done

File: include/iamclient/permservicehandler.hpp (hunk 4 above)
Comment (MykolaSolyanko) [reply include/iamclient/permservicehandler.hpp]: Done

File: tests/iamclient/mocks/certprovider.hpp (hunk 5 above)
Comment (MykolaSolyanko) [reply tests/iamclient/mocks/certprovider.hpp]: Done

File: tests/iamclient/mocks/certprovider.hpp (hunk 6 above)
Comment (MykolaSolyanko) [reply tests/iamclient/mocks/certprovider.hpp]: Done

File: tests/iamclient/mocks/certprovider.hpp (hunk 7 above)
Comment (MykolaSolyanko) [reply tests/iamclient/mocks/certprovider.hpp]: Done

**Review by al1img — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]

**Pull request merged**